pkg/groupsnapshot/controllers/groupsnapshot.go

```diff
@@ -569,7 +569,44 @@ func (m *GroupSnapshotController) handlePostSnap(groupSnap *stork_api.GroupVolum
 }
 
 func (m *GroupSnapshotController) handleFinal(groupSnap *stork_api.GroupVolumeSnapshot) error {
-	// currently nothing to do
+	// Check if user has updated restore namespace
+	latestRestoreNamespacesInCSV := strings.Join(groupSnap.Spec.RestoreNamespaces, ",")
+
+	currentRestoreNamespaces := ""
+	childSnapshots := groupSnap.Status.VolumeSnapshots
+	if len(childSnapshots) > 0 {
+		vsObject, err := k8s.Instance().GetSnapshot(childSnapshots[0].VolumeSnapshotName, groupSnap.GetNamespace())
+		if err != nil {
+			return err
+		}
+
+		childSnapAnnotations := vsObject.Metadata.Annotations
+		if childSnapAnnotations != nil {
+			currentRestoreNamespaces = childSnapAnnotations[snapshot.StorkSnapshotRestoreNamespacesAnnotation]
+		}
+	}
+
+	if latestRestoreNamespacesInCSV != currentRestoreNamespaces {
+		log.GroupSnapshotLog(groupSnap).Infof("Updating restore namespaces for groupsnapshot to: %s",
+			latestRestoreNamespacesInCSV)
+		for _, childSnap := range childSnapshots {
+			vs, err := k8s.Instance().GetSnapshot(childSnap.VolumeSnapshotName, groupSnap.GetNamespace())
+			if err != nil {
+				return err
```

Comment:
This should ignore snapshots that might have been deleted

Reply:
done